core/src/main/java/io/questdb/cairo/wal/WalPurgeJob.java

```diff
@@ -583,4 +413,216 @@ protected boolean runSerially() {
         }
         return false;
     }
+
+    public interface Deleter {
```

Comment:
nit: Deleter is unnecessary, we usually use `FileFacade`. It is an interface already, you can overwrite methods in the tests to track deletions.

Reply:
Yes, I'm aware of this.
I've chosen to introduce this interface because it allows a better separation of the job's data collection and deletion code (which stays in `WalPurgeJob`) from the decision making logic (moved to `WalPurgeJob.Logic`) by removing the need for the logic to know about the DB root.
I've tried to remove as much state from the logic class as possible. For example, the only reason why it still has a `TableToken` is for logging.
Just using the `FileFacade` is certainly possible, but it would make testing more verbose. Let me know if you still want me to get rid of the `Deleter` interface.